cmd/proxy.go

```diff
@@ -46,29 +47,78 @@ hostname and port where it listens.`,
 	}
 )
 
+// ModelRegistryServiceHolder safely holds the model registry service
+type ModelRegistryServiceHolder struct {
```

Comment:
Sorry I' missing the context here. Why is the service holder required?

Reply:
service at the start is nil, in a goroutine that initialize the datastore is set to the correct service so to avoid nil dereference and race conditions I wrapped it in the Holder as this process is async